app/app.go

```diff
@@ -53,99 +43,42 @@ func New(options ...AppOption) *App {
 	return app
 }
 
-func (a *App) InitServer() {
-	a.srv.AppInitializedOnce.Do(func() {
-		a.initEnterprise()
-
-		a.AddConfigListener(func(oldConfig *model.Config, newConfig *model.Config) {
-			if *oldConfig.GuestAccountsSettings.Enable && !*newConfig.GuestAccountsSettings.Enable {
-				if appErr := a.DeactivateGuests(); appErr != nil {
-					mlog.Error("Unable to deactivate guest accounts", mlog.Err(appErr))
-				}
-			}
-		})
-
-		// Disable active guest accounts on first run if guest accounts are disabled
-		if !*a.Config().GuestAccountsSettings.Enable {
-			if appErr := a.DeactivateGuests(); appErr != nil {
+func (s *Server) FinalizeInit(c *request.Context) {
```

Comment:
From looking at the code, it looks like the only reason this is in a separate method is because of tests. Otherwise, the main code just does:

```
server, err := app.NewServer(options...)
if err != nil {
...
}
defer server.Shutdown()
...
server.FinalizeInit(&request.Context{})
```

Would it be fair to say that this code can be refactored some more later on, to make `FinalizeInit` part of `NewServer` itself? That would completely remove any ordering related issues, and have proper dependency injection for service initialization.

Reply:
Yeah, we can move everything inside the `finalizeInit` to `NewServer`, we can create an option for it if it won't be needed in tests.